cmd/lncli/commands.go

```diff
@@ -781,24 +803,111 @@ func create(ctx *cli.Context) error {
 	}
 	fmt.Println()
 
+	// If the passwords don't match, then we'll return an error.
 	if !bytes.Equal(pw1, pw2) {
 		return fmt.Errorf("passwords don't match")
 	}
 
+	// Next, we'll see if the user has 24-word mnemonic they want to use to
+	// derive a seed within the wallet.
+	var (
+		hasMnemonic bool
+		entropy     []byte
+	)
+
+mnemonicCheck:
+	for {
+		fmt.Printf("Do you have an existing 24-word BIP 39 " +
+			"mnemonic you want to use? (Enter y/n): ")
+
+		reader := bufio.NewReader(os.Stdin)
+		answer, err := reader.ReadString('\n')
+		if err != nil {
+			return err
+		}
+
+		fmt.Println()
+
+		answer = strings.TrimSpace(answer)
+		answer = strings.ToLower(answer)
+
+		switch answer {
+		case "y":
+			hasMnemonic = true
+			break mnemonicCheck
+		case "n":
+			hasMnemonic = false
+			break mnemonicCheck
+		}
+	}
+
 	req := &lnrpc.CreateWalletRequest{
 		Password: pw1,
 	}
-	_, err = client.CreateWallet(ctxb, req)
+
+	// If the user *does* have an existing seed they want to use, then
+	// we'll read that in directly from the terminal.
+	if hasMnemonic {
+		// We'll now prompt the user to enter in their 24-word
+		// mnemonic.
+		fmt.Printf("Input your 24-word mnemonic separated by spaces: ")
+		mnemonic, err := terminal.ReadPassword(int(syscall.Stdin))
+		if err != nil {
+			return err
+		}
+		fmt.Println()
+
+		// Now that we have their mnemonic, we'll convert this *back*
+		// into the original entropy which is the format that lnd
+		// requires.
+		entropy, err = bip39.MnemonicToByteArray(string(mnemonic))
+		if err != nil {
+			return err
+		}
+
+		// We'll chop off the final checksum byte as the above function
+		// will error out if the checksum doesn't validate properly.
+		req.Entropy = entropy[:len(entropy)-1]
+	}
+
+	resp, err := client.CreateWallet(ctxb, req)
 	if err != nil {
 		return err
 	}
 
+	// Now if the user didn't specify a source of entropy, then lnd will
+	// return what it used to generate the final seed.
+	generatedEntropy := resp.Entropy
+	if len(generatedEntropy) != 0 {
+		// If this is the case, then we'll convert the entropy into a
+		// mnemonic to display to the user.
+		mnemonic, err := bip39.NewMnemonic(generatedEntropy)
+		if err != nil {
+			return fmt.Errorf("unable to convert entropy to "+
+				"mnemonic (wallet still created): %v", err)
+		}
+
+		fmt.Println("!!!YOU MUST WRITE DOWN THIS SEED TO BE ABLE TO " +
+			"RESTORE THE WALLET!!!\n")
+
+		fmt.Println("Generated mnemonic seed:", mnemonic)
+
+		fmt.Println("\n!!!YOU MUST WRITE DOWN THIS SEED TO BE ABLE TO " +
```

Comment:
Was thinking a little more on how we display the mnemonic to the user in lncli. There's a lot going on there, and think it needs to be more obvious as to what exactly needs to be copied. I would propose printing out the mnemonic in a format similar to text encodings for keys/certs:
```
-----BEGIN LND CIPHER SEED-----
field   soup     wing   canvas
survey  borrow   story  analyst
useless clap     before cheap
giggle  middle   color  visa
cabbage shell    vessel rent
deposit exchange coil   sample
------END LND CIPHER SEED------
```
I think this will greatly reduce the errors people make in writing down the seed. Humans are best at remembering things in 3's and 4's, plus the formatting allows users to check if they're missing a word in each line, instead of hoping they correctly copied all 24 words in a single run-on sentence.

Reply:
I would also say to add numbers...

Japanese users sometimes try to write top to bottom first.

```
-----BEGIN LND CIPHER SEED-----
 1. field    2. soup      3. wing    4. canvas
 5. survey   6. borrow    7. story   8. analyst
 9. useless 10. clap     11. before 12. cheap
13. giggle  14. middle   15. color  16. visa
17. cabbage 18. shell    19. vessel 20. rent
21. deposit 22. exchange 23. coil   24. sample
------END LND CIPHER SEED------
```